src/components/source_editor/LatticeConfigurationDialog.js

```diff
@@ -108,6 +111,22 @@ class LatticeConfigurationDialog extends React.Component {
         const newMaterial = new Material(newMaterialConfig);
         // assert basis is stored in 'crystal' units
         newMaterial.toCrystal();
+
+        // update to non-periodic if asked to do so
+        if (this.state.isNonPeriodic) {
+            const newBasis = new Made.Basis({
+                elements: this.props.material.basis.elements,
+                coordinates: this.props.material.basis.coordinates,
+            });
+            let maxPairwiseDistance = newBasis.maxPairwiseDistance;
+            if (maxPairwiseDistance === 0) {maxPairwiseDistance = 2;}
```

Comment:
This type of logic has to be done in the implementation of the `maxPairwiseDistance` getter itself

Reply:
I will add it there, did not realize we needed it until testing with materials-designer.